src/components/Toolbar.tsx

```diff
@@ -0,0 +1,371 @@
+        />
+        <button
+          className={classes(toolbarButtonClass, pushButtonClass, 'jp-Icon-16')}
+          title={'Push committed changes'}
+          onClick={this._onPushClick}
+        />
+        <button
+          className={classes(
+            toolbarButtonClass,
+            refreshButtonClass,
+            'jp-Icon-16'
+          )}
+          title={'Refresh the repository to detect local and remote changes'}
+          onClick={this._onRefreshClick}
+        />
+      </div>
+    );
+  }
+
+  /**
+   * Renders a repository menu.
+   *
+   * @returns fragment
+   */
+  private _renderRepoMenu() {
+    return (
+      <div className={toolbarMenuWrapperClass}>
+        <button
+          className={toolbarMenuButtonClass}
+          title={`Current repository: ${this.state.repository}`}
+          onClick={this._onRepositoryClick}
+        >
+          <span
+            className={classes(
+              toolbarMenuButtonIconClass,
+              repoIconClass,
+              'jp-Icon-16'
+            )}
+          />
+          <div className={toolbarMenuButtonTitleWrapperClass}>
+            <p className={toolbarMenuButtonTitleClass}>Current Repository</p>
+            <p className={toolbarMenuButtonSubtitleClass}>
+              {PathExt.basename(this.state.repository)}
+            </p>
+          </div>
+          <span
+            className={classes(
+              toolbarMenuButtonIconClass,
+              this.state.repoMenu ? closeMenuIconClass : openMenuIconClass,
+              'jp-Icon-16'
+            )}
+          />
+        </button>
+        {this.state.repoMenu ? null : null}
+      </div>
+    );
+  }
+
+  /**
+   * Renders a branch menu.
+   *
+   * @returns fragment
+   */
+  private _renderBranchMenu() {
+    if (!this.state.repository) {
+      return null;
+    }
+    return (
+      <div className={toolbarMenuWrapperClass}>
+        <button
+          className={toolbarMenuButtonClass}
+          title={`Change the current branch: ${this.state.branch}`}
+          onClick={this._onBranchClick}
+        >
+          <span
+            className={classes(
+              toolbarMenuButtonIconClass,
+              branchIconClass,
+              'jp-Icon-16'
+            )}
+          />
+          <div className={toolbarMenuButtonTitleWrapperClass}>
+            <p className={toolbarMenuButtonTitleClass}>Current Branch</p>
+            <p className={toolbarMenuButtonSubtitleClass}>
+              {this.state.branch}
+            </p>
+          </div>
+          <span
+            className={classes(
+              toolbarMenuButtonIconClass,
+              this.state.branchMenu ? closeMenuIconClass : openMenuIconClass,
+              'jp-Icon-16'
+            )}
+          />
+        </button>
+        {this.state.branchMenu ? (
+          <BranchMenu
+            model={this.props.model}
+            branching={this.props.branching}
+          />
+        ) : null}
```

Comment:
Could be simplified
```ts
        {this.state.branchMenu && (
          <BranchMenu
            model={this.props.model}
            branching={this.props.branching}
          />
        )}
```

Reply:
Same as [here](https://github.com/jupyterlab/jupyterlab-git/pull/518#discussion_r370911399).